src/common_interface/solve.jl

```diff
@@ -111,13 +111,13 @@ function DiffEqBase.__init(
         alg_code = CV_ADAMS
     end
 
-    if Method == :Newton
```

Comment:
@YingboMa note that this will likely be a spot that's broken due to nonlinear solver changes.

Reply:
Would it make sense to make structs for the linear/non-linear solver so we can set options for those? It seems that the current implementation doesn't interface the solver's parameter settings.

Reply:
Those parameter settings didn't exist before. So yeah, we can make those structs and improve the interface now, but let's make that a follow up.

Reply:
Ok, noted. We can help with that later since it is critical for us